job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java

```diff
@@ -169,15 +169,17 @@ private static void migrate(MigrateCommand command,
     do {
       retry = false;
       try (FileInStream in = fileSystem.openFile(new AlluxioURI(source), openFileOptions);
-          FileOutStream out = fileSystem.createFile(destinationURI, createOptions)) {
+           FileOutStream out = fileSystem.createFile(destinationURI, createOptions)) {
         try {
           IOUtils.copyLarge(in, out, new byte[8 * Constants.MB]);
         } catch (Throwable t) {
+          // Since we only catch CancelledException here if we get interrupted, so we want to keep
```

Comment:
@ggezer The usage here is rather unconventional, but it is for a good reason. Changed the comment to make it more clear. We are explicitly clearing the interrupted flag because we are on our way out of this function/future and performing cleanup tasks. The interrupted flag will cause  `fileSystem.delete` to throw InterruptedException otherwise.   It is an noop if the function exited for  IOException reasons

Reply:
Should we reset interrupted state after deleting the file then?(Only if it was interrupted of course since we're catching all throwables here)
I just wanna make sure, this doesn't break another logic later that depends on interruption.

Reply:
In TaskExecutor line 91, we interrupt the thread again if catch related Exception which is thrown after delete.

Reply:
if an InterruptedException was thrown, then the flag would have been reset when it was thrown. The key issue here is Future throws a CancelledException instead of InterruptedException, and does not clear interrupt flag.

Reply:
huh. interesting behavior.